### PR TITLE
sdk/js: handling valid exceptions

### PR DESCRIPTION
**Pull request by Abhishekkochar**
fix: #3190

### CODE REVIEW

**Comment by Abhishekkochar** [comment]
@kcsongor, Please have a look.

File: sdk/js/src/token_bridge/__tests__/algorand-integration.ts [hunk 1]
```diff
@@ -127,7 +127,11 @@ describe("Algorand tests", () => {
             vaaBytes
           );
         } catch (e) {
-          success = false;
+          if(e instanceof Error && e.message.includes("wrapped asset already exists")){
+            success = true;
```
Comment (kcsongor) [inline sdk/js/src/token_bridge/__tests__/algorand-integration.ts:131]: shouldn't this be the other way around?

File: sdk/js/src/token_bridge/__tests__/algorand-integration.ts [hunk 2]
```diff
@@ -370,7 +374,12 @@ describe("Algorand tests", () => {
             vaaBytes
           );
         } catch (e) {
-          success = false;
+          //@notice
+          if(e instanceof Error && e.message.includes("wrapped asset already exists")){
+            success = true;
```
Comment (kcsongor) [inline sdk/js/src/token_bridge/__tests__/algorand-integration.ts:379]: this one too

File: sdk/js/src/token_bridge/__tests__/aptos-integration.ts [hunk 3]
```diff
@@ -420,6 +425,11 @@ describe("Aptos SDK tests", () => {
       await createWrappedOnEth(ethTokenBridge, recipient, attestVAA);
     } catch (e) {
       // this could fail because the token is already attested (in an unclean env)
+      if(e instanceof Error && e.message.includes("wrapped asset already exists")){
```
Comment (kcsongor) [inline sdk/js/src/token_bridge/__tests__/aptos-integration.ts:428]: does the error message actually contain this?

**Review by kcsongor — CHANGES_REQUESTED** [review]
Did you run these tests locally?

File: sdk/js/src/token_bridge/__tests__/algorand-integration.ts (hunk 1 above)
Comment (Abhishekkochar) [reply sdk/js/src/token_bridge/__tests__/algorand-integration.ts:131]: Oh yes. Fixed. Will push soon.

**Comment by Abhishekkochar** [comment]
@kcsongor, Can you please confirm if this is correct [algogrand/package](https://github.com/wormhole-foundation/wormhole/blob/main/algorand/package.json#L14)?
There is no test.ts file in test dir.

**Comment by Abhishekkochar** [comment]
> Did you run these tests locally?

I have tried to run them. There is an error when running alogrand tests?

**Comment by evan-gray** [comment]
This error case is only hit when the tests are run a second time on an existing tilt instance. This never happens in CI and is only a hinderance to a developer trying to run the tests without cycling the corresponding chains. Given that the subsequent test cannot pass if the token was not properly attested, the existing code is adequate for CI. So, for the benefit of devs in this repo, it seems a more prudent thing would be to skip the test step altogether if the token was already attested.

**Pull request closed without merging**